streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/StateStoreNode.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals.graph;
+
+import org.apache.kafka.streams.processor.internals.InternalTopologyBuilder;
+import org.apache.kafka.streams.state.StoreBuilder;
+
+public class StateStoreNode extends StreamsGraphNode {
+
+    protected final StoreBuilder storeBuilder;
+    private final String[] processorNames;
+
+    public StateStoreNode(final StoreBuilder storeBuilder,
+                          final String... processorNames) {
+        super(storeBuilder.toString(), false);
+
+        this.storeBuilder = storeBuilder;
+        this.processorNames = processorNames;
+    }
+
+    @Override
+    public void writeToTopology(final InternalTopologyBuilder topologyBuilder) {
+
+        if (processorNames != null && processorNames.length > 0) {
+            topologyBuilder.addStateStore(storeBuilder, processorNames);
+        } else {
+            topologyBuilder.addStateStore(storeBuilder);
```

Comment:
I am wondering if we should throw for this case? Note, that in the original code, we need to allow this, ie, add store and later pass store to `transform(...)` etc.

However, because `build()` is called later, the names of all processors the store is attached to is know already -- and if a store is not used, we might want to report an error?